### ISSUE TITLE
Doesn't compile on non-uwp Windows

### ISSUE THREAD

**Issue by udoprog** [aspenluxxxy/nanorand-rs#3]
So I currently get this when trying to compile nanorand on my Windows box:

```
  = note: libnanorand-3df99cf1d4f057c6.rlib(nanorand-3df99cf1d4f057c6.nanorand.3usfjb36-cgu.13.rcgu.o) : error LNK2019: unresolved external symbol BCryptGenRandom referenced in function _ZN8nanorand7entropy7windows19entropy_from_system17heb014656c23127ddE
```

This appears to be due to this being a non-uwp platform. Note that this happens even if the `getrandom` feature is enabled, because [the `windows` entryopy module is still included with `#[cfg(windows)]`](https://github.com/aspenluxxxy/nanorand-rs/blob/master/src/entropy/mod.rs#L9), causing it to be linked in.

[This appears to be the `cfg`](https://github.com/rust-random/getrandom/blob/master/src/lib.rs#L207) that should be checked for to make sure `BCryptGenRandom` is available.
```rust
#[cfg(all(windows, target_vendor = "uwp"))]
```

OTOH, all the necessary platform detection already appears to be implemented in `getrandom`, so maybe we should just be relying on it instead? [There's also some error handling being done](https://github.com/rust-random/getrandom/blob/master/src/windows_uwp.rs#L36) which does not seem to be reproduced in nanorand as well.

**Comment by aspenluxxxy** [comment]
What version of Windows are you on? It should work on anything Vista/Server 2008 and up.

Also, the reason `getrandom` isn't default is because I want there to be an option to compile without any deps or derives.

**Comment by aspenluxxxy** [comment]
Try commit 7076fbd7322f40730524fa99574b0b7d5de457ab.

**Comment by udoprog** [comment]
> What version of Windows are you on? It should work on anything Vista/Server 2008 and up.
![image](https://user-images.githubusercontent.com/111092/93956663-0074f800-fd53-11ea-86cc-3c854eecfd81.png)

While probably available on the system (I have the Windows SDK enabled), it requires `bcrypt.lib` to be linked. And it's probably just the case that the `uwp` target of rustc [links it in by default](https://github.com/rust-lang/rust/blob/master/library/std/build.rs#L51). So filtering the availability by `target_vendor = "uwp"` might be appropriate.

> Also, the reason `getrandom` isn't default is because I want there to be an option to compile without any deps or derives.
All right. But when `getrandom` is enabled, the `windows` module should probably not be included at all I'm guessing?

**Comment by udoprog** [comment]
> Try commit [7076fbd](https://github.com/aspenluxxxy/nanorand-rs/commit/7076fbd7322f40730524fa99574b0b7d5de457ab).

Yep, that should work. We raced past each other :)

**Comment by aspenluxxxy** [comment]
Should be fixed in nanorand 0.4.3.
